examples/hybrid_search.py

```diff
@@ -0,0 +1,122 @@
+import turbopuffer as tpuf
+
+
+# Helper to convert the results to a dictionary with ranks
+def results_to_ranks(results, reverse=False):
+    if reverse:
+        results = sorted(results, key=lambda item: item.dist, reverse=True)
+    return {item.id: rank for rank, item in enumerate(results, start=1)}
+
+
+# Fuses two search result sets together into one
+# Uses reciprocal rank fusion
+def rank_fusion(bm25_results, vector_results, k=60):
```

Comment:
I think we may want to inline/copy-paste this in a few places, so I think we need 1 top-level comments and then the function to be as small as possible, e.g. something alone the lines of this (not tested):

```python
def results_to_ranks(results, reverse=False):
    return {item.id: rank for rank, item in enumerate(sorted(results, key=lambda item: item.dist, reverse=reverse), start=1)}

def rank_fusion(bm25_results, vector_results, k=60):
    bm25_ranks, vector_ranks = results_to_ranks(bm25_results), results_to_ranks(vector_results, reverse=True)
    scores = {doc_id: 1 / (k + bm25_ranks.get(doc_id, k)) + 1 / (k + vector_ranks.get(doc_id, k))
              for doc_id in set(bm25_ranks) | set(vector_ranks)}
    return [{"id": doc_id, "score": score} for doc_id, score in sorted(scores.items(), key=lambda item: item[1], reverse=True)]
```